src/reformulate.jl

```diff
@@ -1,82 +1,454 @@
+# disjunctions
+function _reformulate_disjunctions(model::JuMP.Model, method::AbstractReformulationMethod)
+    for (_, disj) in _disjunctions(model)
+        _reformulate_disjunction(model, disj, method)
+    end
+end
 
-Reformulate disjunction.
-"""
-function reformulate_disjunction(m::Model, disj...; bin_var, reformulation, param)
-    #check disj
-    disj = [check_constraint!(m, constr) for constr in disj]#check_disjunction!(m, disj)
-    #get original variable refs and variable names
-    vars = setdiff(all_variables(m), m[bin_var])
-    var_names = unique(Symbol.([split("$var","[")[1] for var in vars]))
-    if !in(:gdp_variable_refs, keys(object_dictionary(m)))
-        @expression(m, gdp_variable_refs, vars)
+# disjuncts
+function _reformulate_disjunction(model::JuMP.Model, disj::ConstraintData{T}, method::Union{BigM,Indicator}) where {T<:Disjunction}
+    for d in disj.constraint.disjuncts
+        _reformulate_disjunct(model, d, method)
     end
-    if !in(:gdp_variable_names, keys(object_dictionary(m)))
-        @expression(m, gdp_variable_names, var_names)
+end
+function _reformulate_disjunction(model::JuMP.Model, disj::ConstraintData{T}, method::Hull) where {T<:Disjunction}
+    disj_vrefs = _get_disjunction_variables(model, disj)
+    _update_variable_bounds.(disj_vrefs)
+    hull = _Hull(method.value, disj_vrefs)
+    for d in disj.constraint.disjuncts #reformulate each disjunct
+        _disaggregate_variables(model, d, disj_vrefs, hull) #disaggregate variables for that disjunct
+        _reformulate_disjunct(model, d, hull)
     end
-    #run reformulation
-    if reformulation == :hull
-        disaggregate_variables(m, disj, bin_var)
-        sum_disaggregated_variables(m, disj, bin_var)
+    for vref in disj_vrefs #create sum constraint for disaggregated variables
+        _aggregate_variable(model, vref, hull)
     end
-    reformulate_disjunction(disj, bin_var, reformulation, param)
+end
 
-    #show new constraints as a Dict
-    new_constraints = Dict{Symbol,Any}(
-        Symbol(bin_var,"[$i]") => disj[i] for i in eachindex(disj)
-    )
-    new_constraints[Symbol(bin_var,"_XOR")] = constraint_by_name(m, "XOR(disj_$bin_var)")
-    if reformulation == :hull
-        for var in m[:gdp_variable_refs]
-            agg_con_name = "$(var)_$(bin_var)_aggregation"
-            new_constraints[Symbol(agg_con_name)] = constraint_by_name(m, agg_con_name)
-        end
+# individual disjuncts
+function _reformulate_disjunct(model::JuMP.Model, ind_idx::LogicalVariableIndex, method::AbstractReformulationMethod)
+    #reformulate each constraint and add to the model
+    bv_idx = _indicator_to_binary(model)[ind_idx]
+    bvref = JuMP.VariableRef(model, bv_idx)
+    for cidx in _indicator_to_constraints(model)[ind_idx]
+        cdata = _disjunct_constraints(model)[cidx]
+        reformulate_disjunct_constraint(model, cdata.constraint, bvref, method)
     end
-    return new_constraints
+end
 
```

Comment:
We should add an extension point with a docstring to explain the API:
```julia
"""
my docstring
"""
function reformulate_disjunct_constraint end
```
I also think we should make `_reformulate_disjunction` public as well.